### PR TITLE
Add clair to drone

### PR DESCRIPTION
**Pull request by andriisoldatenko**
fix https://github.com/astronomer/astronomer/issues/448
i used `shellcheck bin/clair-scan` to check bash style correctness. 
related PR https://github.com/astronomer/ap-build/pull/1

Before merge need to apply:
- [x] Done
```diff
diff --git a/.drone.yml b/.drone.yml
index 3c5737b..e726804 100644
--- a/.drone.yml
+++ b/.drone.yml
@@ -88,15 +88,19 @@ pipeline:
       branch: [ master, release-* ]

   scan-platform:
-    image: asoldatenko/ap-build:latest
+    group: push
+    image: astronomerio/ap-build:0.1.2
     commands:
       - make scan-platform
     environment:
-      - ASTRONOMER_VERSION=0.10.0
+      - ASTRONOMER_VERSION=0.0.0
       - ASTRONOMER_REF=${DRONE_BRANCH}
       - BUILD_NUMBER=${DRONE_BUILD_NUMBER}
     volumes:
       - /var/run/docker.sock:/var/run/docker.sock
+    when:
+      event: push
+      branch: [ master, release-* ]

   # trigger:
```
- Success build https://drone.astronomer.io/astronomer/astronomer/1140/6

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -79,6 +88,17 @@ pipeline:
       event: push
       branch: [ master, release-* ]
 
+  scan-platform:
+    image: astronomerio/ap-build:0.1.1
+    commands:
+      - make scan-platform
+    environment:
+      - ASTRONOMER_VERSION=0.10.0
```
Comment (ashb) [inline .drone.yml]: What's this used for?

File: bin/clair-scan [hunk 2]
```diff
@@ -1,43 +1,26 @@
...
-                          "ap-nginx-es"
-                          "ap-pgbouncer"
-                          "ap-pgbouncer-exporter"
-                          "ap-prisma"
-                          "ap-prometheus"
-                          "ap-redis"
-                          "ap-registry"
-                          "ap-statsd-exporter")
+if [[ ! -f $(which clair-scanner) ]]; then
+    wget https://github.com/arminc/clair-scanner/releases/download/v12/clair-scanner_linux_amd64 -O /usr/local/bin/clair-scanner
```
Comment (ashb) [inline bin/clair-scan]: Instead of doing this add the binary in to the ap-build image and depend on that new tag?

File: .drone.yml (hunk 1 above)
Comment (andriisoldatenko) [reply .drone.yml]: Need remove it

File: .drone.yml [hunk 3]
```diff
@@ -6,6 +6,15 @@ pipeline:
   # Update platform Dockerfiles to point to extend ap-base image
   # Update airflow onbuild images to extend the corresponding base image
```
Comment (ashb) [inline .drone.yml]: You've split this comment from the block it applied to (`update-tags`)

File: .drone.yml (hunk 1 above)
Comment (ashb) [reply .drone.yml]: In the `build-platform` stage we have this as `0.0.0` in the this file and it is over-written by the `bin/release-platform` script. We should probably the same here.

File: bin/clair-scan (hunk 2 above)
Comment (andriisoldatenko) [reply bin/clair-scan]: done,

File: bin/clair-scan [hunk 4]
```diff
@@ -1,43 +1,43 @@
...
+    echo "Please install https://github.com/arminc/clair-scanner"
+    exit 1
+fi
+
+IMAGE=${PUSH_IMAGE}:${PUSH_TAGS}
+RESULTS_FILE="${IMAGE//[\/\-:\.]/_}_results.txt"
+#HOST_IP="$(hostname -I | awk '{print $1}')" || true
+CLAIR_HOST_IP=${CLAIR_HOST:-clair_local_scan}
+
+docker pull "${IMAGE}" &> /dev/null
```
Comment (ashb) [inline bin/clair-scan]: I'm not 100% sure when in the pipeline this is run, but we want to run this against the image we have build before it is published to dockerhub (i.e. before the `push-platform` or `push-airflow` stages of the drone pipeline has run, and against the local images.

Comment (andriisoldatenko) [reply bin/clair-scan]: it uses same docker socket as during build image, that's why it's local images.

**Comment by ashb** [comment]
```
{"Event":"pgsql: could not open database: dial tcp: lookup postgres on 127.0.0.11:53: no such host","Level":"fatal","Location":"main.go:96","Time":"2019-08-28 14:56:07.268517"}
```

but the step was marked as success?

File: .drone.yml [hunk 5]
```diff
@@ -1,4 +1,12 @@
 pipeline:
+  wait_for_db:
+    image: astronomerio/ap-build:0.1.1
+    commands:
+      - while ! nc -w 1 -z postgres 5432; do sleep 1; done
+
+  clair_local_scan:
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply .drone.yml]: it does wait for db, we need wait until service:
```
services:
  postgres:
    image: arminc/clair-db
    detach: true
    pull: true
```
running

[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply .drone.yml]: from github issue in drone repo, i found only this approach will work, there is no way to wait 1 service for another, at least  i don't know how.

[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply .drone.yml]: for history from slack:
clair consists of 3 parts:
- database arminc/clair-db -> postgres
- web application (api endpoint) clair_local_scan
- cli app (clair-scanner) (edited)

**Comment by andriisoldatenko** [comment]
@ashb @schnie PTAL

**Pull request merged**